.github/workflows/pull_request.yaml

```diff
@@ -13,8 +13,25 @@ jobs:
           go-version: 1.17
       - name: Check out source code
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: dangoslen/changelog-enforcer@v3
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
       - name: Run tests
         run: make
+      - name: Run Lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: v1.46.2
+          skip-pkg-cache: true
+          only-new-issues: true
+          github-token: ${{ github.token }}
```

Comment:
So I use this approach mainly because it's what [Golang-ci recommends](https://golangci-lint.run/usage/install/#github-actions), and it enhances the developer experience by doing things like adding the annotations of the troublesome lines right to the pull request. In terms of the configuration of the linters, since that's all stored in `.golangci-lint.yaml`, I _think_ it should be consistent between developers running `make lint` and what the CI executes (in that you won't see failures in one place that don't occur in the other).

Let me make a separate PR that updates the README with some guidelines about stuff you should do locally (like `vet`, `test-fmt`, and `lint`), which should help. I was considering implementing Git hooks for `lint` and `test-fmt` actually as well. Make sense? If not, let's chat in another issue, because I value your experience here!